### ISSUE TITLE
panic when inserting into table with multi-column vindex

### ISSUE THREAD

**Issue by slanning** [vitessio/vitess#4452]
#### Overview of the Issue

I'm getting a panic whenver I try to insert into a table that has two columns in a vindex.
Edit/update: This only happened after we split into two shards; was okay unsharded.

#### Reproduction Steps

1. Deploy the following `vschema` (hopefully fixed now after @demmer's comment):

    ```javascript
   {
     "sharded": true,
     "vindexes": {
       "hash": {
         "type": "hash"
       }
     },
     "tables": {
       "TestMultiCol": {
         "column_vindexes": [
             {
                 "columns": ["id1", "id2"],
                 "name": "hash"
             }
         ]
       }
     }
   }
    ```

2. Deploy the following `schema`:

    ```sql
    create table TestMultiCol (id1 int unsigned not null, id2 int unsigned not null, primary key (id1,id2)) engine=InnoDB;;
    ```

3. Run `insert into TestMultiCol (id1, id2) values (1,1);`

4. The mysql client loses its connection to the server. Output in vtgate log:
```
E1213 13:31:34.027127   53837 server.go:257] mysql_server caught panic:
runtime error: index out of range
/usr/lib/golang/src/runtime/panic.go:513 (0x42da38)
/usr/lib/golang/src/runtime/panic.go:44 (0x42c879)
$builddir/src/vitess.io/vitess/go/vt/vtgate/engine/insert.go:398 (0x9b0e6a)
$builddir/src/vitess.io/vitess/go/vt/vtgate/engine/insert.go:217 (0x9aea0e)
$builddir/src/vitess.io/vitess/go/vt/vtgate/engine/insert.go:167 (0x9ae2a1)
$builddir/src/vitess.io/vitess/go/vt/vtgate/executor.go:301 (0xa1b60d)
$builddir/src/vitess.io/vitess/go/vt/vtgate/executor.go:205 (0xa19de5)
$builddir/src/vitess.io/vitess/go/vt/vtgate/executor.go:126 (0xa19a60)
$builddir/src/vitess.io/vitess/go/vt/vtgate/vtgate.go:304 (0xa39251)
$builddir/src/vitess.io/vitess/go/vt/vtgate/plugin_mysql_server.go:154 (0xa29b83)
$builddir/src/vitess.io/vitess/go/mysql/conn.go:730 (0x81d6be)
$builddir/src/vitess.io/vitess/go/mysql/server.go:436 (0x82d27c)
/usr/lib/golang/src/runtime/asm_amd64.s:1333 (0x45d960)
```

#### Binary version

Various builds from at least the past few weeks.

**Comment by slanning** [comment]
Ok, thanks. Guess I don't understand yet what to do with multi-column primary keys yet.

**Comment by demmer** [comment]
For slack we pick one column to be the sharding vindex and the rest as part of the pk.

As long as one column has enough cardinality this should avoid hot spots.

**Comment by sougou** [comment]
Fixed via #5509.
